File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 5 - Multiple Linear Regression/my_multiple_linear_regression.py
# Multipple Linear Regression

# Importing the Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('50_Startups.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 4].values

# Handling categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder = LabelEncoder()
X[:, 3] = labelencoder.fit_transform(X[:, 3])
onehotencoder = OneHotEncoder(categorical_features = [3])
X = onehotencoder.fit_transform(X).toarray()

# Avoiding the Dummy variable trap
X = X[:, 1:]

# Splitting the training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# The library is going to take care of the Feature Scaling

# Fitting the Multiple Linear Regression over the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Prediction over the test set
y_pred = regressor.predict(X_test)

File: Machine Learning A-Z Template Folder/Part 2 - Regression/Section 5 - Multiple Linear Regression/my_multiple_linear_regression.R
# Importing the dataset
dataset = read.csv('50_startups.csv')

# Handling categorical data
dataset$State = factor(dataset$State,
                                 levels = c('New York', 'California', 'Florida'),
                                 labels = c(1, 2, 3))

# Splitting the training and test sets
library(caTools)
set.seed(123)
split = sample.split(dataset$Profit, SplitRatio = 0.8)
training_set = subset(dataset, split == T)
test_set = subset(dataset, split == F)

# Fitting Multiple Linear Regression to the training set
regressor = lm(formula = Profit ~ ., data = training_set)
summary(regressor)

# Predicting the test set results
pred = predict(regressor, newdata = test_set)
pred
test_set[,5]
